mypyc/emitwrapper.py

```diff
@@ -36,14 +36,24 @@ def make_format_string(func_name: str, groups: List[List[RuntimeArg]]) -> str:
     return '{}:{}'.format(main_format, func_name)
 
 
-def generate_wrapper_function(fn: FuncIR, emitter: Emitter) -> None:
+def generate_wrapper_function(fn: FuncIR,
+                              emitter: Emitter,
+                              source_path: str,
+                              module_name: str) -> None:
     """Generates a CPython-compatible wrapper function for a native function.
 
     In particular, this handles unboxing the arguments, calling the native function, and
     then boxing the return value.
     """
     emitter.emit_line('{} {{'.format(wrapper_function_header(fn, emitter.names)))
 
+    globals_static = emitter.static_name('globals', module_name)
+    traceback_code = 'CPy_AddTraceback("%s", "%s", %d, %s);' % (
+        source_path.replace("\\", "\\\\"),
+        fn.traceback_name or fn.name,
```

Comment:
Is it possible to move this logic to `.name` (which is a property)?

Reply:
I think I'd rather be explicit about it here. But I'll add a comment explaining why we are doing a traceback at all when there is no traceback_name, even though we don't in other cases.